Sources/Logging/Logging.swift

```diff
@@ -38,21 +38,61 @@ import WASILibc
 /// ```swift
 /// logger.info("Hello World!")
 /// ```
+
 public struct Logger {
+    /// Storage class to hold the label and log handler
+    @usableFromInline
+    internal final class Storage {
+        @usableFromInline
+        var label: String
+        @usableFromInline
+        var handler: LogHandler
+        @usableFromInline
+        init(label: String, handler: LogHandler) {
+            self.label = label
+            self.handler = handler
+        }
+    }
+    
+    @usableFromInline
+    internal var _storage: Storage
     @usableFromInline
-    var handler: LogHandler
+    internal var label: String {
+        get {
+            return _storage.label
+        }
+        set {
+            if !isKnownUniquelyReferenced(&_storage) {
+                _storage = Storage(label: newValue, handler: _storage.handler)
```

Comment:
it is usually beneficial to create a `func copy() -> Storage` on storage that just copies all variables over then you can write this as

```
[...]
self._storage = self._storage.copy()
self._storage.label = newValue
[...]
```

Also missing explicit `self.`